src/source/Signaling/ChannelInfo.c

```diff
@@ -308,3 +340,95 @@ PCHAR getStringFromChannelRoleType(SIGNALING_CHANNEL_ROLE_TYPE type)
 
     return typeStr;
 }
+
+// https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-how-iam.html#kinesis-using-iam-arn-format
+// Example: arn:aws:kinesisvideo:region:account-id:channel/channel-name/code
+STATUS validateKvsSignalingChannelArnAndExtractChannelName(PChannelInfo pChannelInfo, PUINT16 pStart, PUINT16 pNumChars)
+{
+    UINT16 arnLength, currPosIndex = 0, channelNameLength = 0, channelNameStart = 0;
+    PCHAR partitionEnd, regionEnd, channelNameEnd;
+    PCHAR currPos;
+    UINT8 accountIdStart = 1, accountIdEnd = 12;
+    UINT8 timeCodeStart = 0, timeCodeEnd = 12;
+
+    if (pChannelInfo != NULL && pChannelInfo->pChannelArn != NULL) {
+        arnLength = STRNLEN(pChannelInfo->pChannelArn, MAX_ARN_LEN);
+
+        if (arnLength >= SIGNALING_CHANNEL_ARN_MIN_LENGTH) {
```

Comment:
Wondering if we can convert this entire nested if block into switch cases. Why not run a while loop and then use enums in a switch block to parse? It would be cleaner if it can be achieved and also maintainable.
typedef enum {
    PARSE_START = 0,
    PARSE_ARN_PREFIX,
    PARSE_PARTITION,
    PARSE_SERVICE,
    PARSE_REGION,
    PARSE_ACCOUNT_ID,
  ......
    PARSE_DONE
} ARN_PARSE_STATE;

We do a similar thing for ICE candidate parsing from incoming SDP:
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/develop/src/source/Ice/IceAgent.c#L9-L19
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/develop/src/source/Ice/IceAgent.c#L359-L418